Components/AllEventPage/EventBox.tsx

```diff
@@ -1,29 +1,89 @@
+import { Alert, AlertTitle } from "@mui/material";
+import { useEffect, useState } from "react";
 import classes from "./EventBox.module.css";
 import EventBoxContent from "./EventBoxContent";
 import EventElement from "./EventElement";
 
+
+interface Organizer {
+  shortName: string;
+  displayName: string;
+}
+
+
+interface Registration {
+  link: string;
+  type: string;
+}
+
+interface Event {
+  id: string;
+  subtitle: string;
+  displayName: string;
+  description: string;
+  endAt: number;
+  startAt: number;
+  organizers: Organizer[];
+  registration: Registration;
+}
+
+interface EventResponse {
+  isError: boolean;
+  data: Event[];
+}
+
+const ErrorAlert = (props: { errorMessage: string }) => {
+  return (
+    <Alert severity="error">
+      <AlertTitle>Error while fetching events!</AlertTitle>
+      {props.errorMessage}
+    </Alert>
+  );
+};
+
+const epochToString = (epoch: number) => new Date(epoch).toLocaleString("en-IN");
+
 function EventBox() {
+  const [events, setEvents] = useState<Event[]>([]);
+  const [errorMessage, setErrorMessage] = useState<string>("");
+
+  useEffect(() => {
+    fetch('https://asia-south1-nsccpccoe.cloudfunctions.net/events')
+      .then(res => {
+        if (!res.ok) {
+          throw res.statusText;
+        }
+        return res.json();
+      })
+      .then((eventResponse: EventResponse) => {
+        if (eventResponse.isError) {
+          setErrorMessage("Unknown error occured while fetching event details");
```

Comment:
API also provide `errorCode` and `errorMessage`. You can use errorCode to identify error types and you can directly display `errorMessage` to user.

Refer the below API Interfaces:

```
export interface CustomResult<T> {
    isError: false
    data: T
}
```
```
export interface CustomError {
    isError: true
    errorCode: string
    errorMessage: string
}
```

Reply:
`errorCode` and `errorMessage` are missing in api response.
![image](https://user-images.githubusercontent.com/56870381/212562446-58f4db43-ad63-4483-b88b-fe56088b9d41.png)

Reply:
`errorCode` and `errorMessage` will be visible only when `isError` will be `true`

Reply:
`typeof respone = CustomResult<T> | CustomError`

Reply:
Thanks for clarifying!